src/main/java/com/thebois/models/world/World.java

```diff
@@ -178,10 +189,18 @@ public boolean createStructure(final StructureType type, final Position position
      *
      * @return Whether the structure was built.
      */
-    public boolean createStructure(final StructureType type, final int x, final int y) {
+    public boolean tryCreateStructure(final StructureType type, final int x, final int y) {
         final Position position = new Position(x, y);
+        StructureType typeToCreate = type;
+        if (!isTownHallPlaced()) {
+            typeToCreate = StructureType.TOWN_HALL;
+        }
+        else if (type.equals(StructureType.TOWN_HALL)) {
+            typeToCreate = DEFAULT_STRUCTURE_TYPE;
```

Comment:
No need for a default structure type. A type is already passed in from the controller in the `type` parameter.

Reply:
Also, could you clarify why this else if-branch is needed? If the town hall is already placed, then the asked for `type` should be what is placed, no need to have a separate branch setting that, right?

At least, an else-branch should be enough, right?

Reply:
Ah, I see. Is it possible for the player to try to place the townhall even after it's been placed once?

Reply:
Now placeStructureController is responsible for sending the correct `StructureType` to `World`